sktime/utils/_testing/deep_equals.py

```diff
@@ -112,6 +168,14 @@ def ret(is_equal, msg="", string_arguments: list = None):
         else:
             return ret(x.equals(y), ".df_equals, x = {} != y = {}", [x, y])
     elif isinstance(x, pd.Index):
+        if hasattr(x, "dtype") and hasattr(y, "dtype"):
```

Comment:
This lines look almost like the code in line 117 to 130. perhaps might be worth to factor this out into an additional method. 


Furthermore, should the following check be the same as the index checks of the dataframe?

Reply:
yes, I am secretly planning the "factor out" in an upwards refactor based on https://github.com/sktime/skbase/pull/243, i.e., plugins for equality checking for different types.

Reply:
> Furthermore, should the following check be the same as the index checks of the dataframe?

At some point I tried to recurse the latter actually to the former, but that fails since `deep_equals` is stricter than `xix.equals(yix)`, which in turn is stricter than `np.all(xix == yix)`, as discussed above.

The reason why `deep_equals` is stricter is that it first checks the python type, but `pandas` `Index.equals` and `DataFrame.equals` allow indices with different python type to be equal, e.g., `RangeIndex`, and `Index` with integer values.

At this point, I'm scared of any minimal change to this apparently working code.

Reply:
(obviously, I blame `pandas`)

Reply:
Thanks for the explanation